### ISSUE TITLE
GroupedList should be represented as a tree/treeitems  (as well as the keyboard navigation)

### ISSUE THREAD

**Issue by jspurlin** [OfficeDev/office-ui-fabric-react#2084]
### Bug Report

- __Package version(s)__: office-ui-fabric-react_v4.12.0

#### Priorities and help requested (not applicable if asking question):

Requested priority: Normal

Products/sites affected: all consumers of GroupedList

#### Describe the issue:

GroupedList is currently marked up as nested list/listitem but the UI representation of the data is [tree](https://www.w3.org/TR/wai-aria/roles#tree)/[treeitem](https://www.w3.org/TR/wai-aria/roles#treeitem) (as well as the treeitem elements having aria-level, aria-posinest, aria-setsize, aria-expanded, aria-haspopup, aria-checked). The markup should be updated to align with the UI. On top of that, the keyboarding behavior needs to be updated to align with tree navigation. Currently the GroupedList doesn't seem to be able to be collapsed/expanded via the keyboard. See: https://www.w3.org/TR/2009/WD-wai-aria-practices-20090224/#TreeView for the expected behavior

**Comment by micahgodbolt** [comment]
@jspurlin i see that #1975 was merged. Does that solve this issue? If so, can you close it?

**Comment by jspurlin** [comment]
@micahgodbolt  no, #1975 removed some extra/unneeded markup, but this bug is tracking fixing the markup of that control to be represented more correctly as tree/treeitem (which is a lot more work)
